File: pset1/cash/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main()
{
    // Asks for user input (positive real numbers) 
    float dollars;
    do
    {
        dollars = get_float("Change owed : ");
    }
    while (dollars < 0);
    
    // Rounds up the decimal places (C overflow)
    int cents = round(dollars * 100);

    // Math part of source code
    int tfvc = cents / 25;   
    int tenc = (cents % 25) / 10;
    int fivc = ((cents % 25) % 10) / 5;
    int onec = ((cents % 25) % 10) % 5;

    // Displays the minimum amount of coins needed to give the inputed change
    printf("%d\n", tfvc + tenc + fivc + onec);
}

File: pset6/cash.py
# Calculate the minimum number of coins required to give a user change

from cs50 import get_float
from math import floor


def main():

    # User input for change owed in dollars (require positve)
    while True:
        dollars = get_float("Change owed: ")
        if dollars < 0:
            continue
        else:
            break

    # Algorithm
    cents = round(dollars * 100)
    quarters = floor(cents / 25)
    dimes = floor((cents - quarters * 25) / 10)
    nickels = floor((cents - quarters * 25 - dimes * 10) / 5)
    pennies = floor((cents - quarters * 25 - dimes * 10 - nickels * 5))

    print(f"{quarters + dimes + nickels + pennies}")


main()
